build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java

```diff
@@ -447,8 +447,10 @@ private boolean isFinished(BuildSetTask buildSetTask) {
     private void handleErroneousFinish(BuildTask failedTask) {
         BuildSetTask taskSet = failedTask.getBuildSetTask();
         if (taskSet != null) {
+            log.debug("Finishing tasks in set {}, after failedTask {}.", taskSet, failedTask);
             taskSet.getBuildTasks().stream()
                     .filter(t -> isDependentOn(failedTask, t))
+                    .filter(t -> !t.getStatus().isCompleted())
```

Comment:
Test scenario I added has the dependency chain.
A->B->C, where C is failing and B fall into infinite loop.

Yes that extra filter is a real change.